node/actors/internal_nats_server.go

```diff
@@ -25,23 +45,64 @@ type agentCredential struct {
 	nkey         nkeys.KeyPair
 }
 
+type configTemplateData struct {
+	Credentials       map[string]*credentials
+	Connections       map[string]*nats.Conn
+	NexHostUserPublic string
+	NexHostUserSeed   string
+}
+
+type credentials struct {
```

Comment:
I still think this is redundant and we should consider reusing `agentCredential`.

Reply:
I still disagree. I don't like the idea of reusing a data structure for internal functional code for stuff used solely as the data source for a template generator. It might seem like a nitpick, but I don't like bleeding responsibilities like this. If we combine, then one purpose will make no use of two fields and the other purpose will make no use of one field. Keeping them separate means:
* it's clear that these data structures are meant for different purposes
* if we want to add additional stuff to the template generating struct later, it's clear how to do that and we don't have to disentangle them.
* we don't have the dangling orphan fields. While it might be obvious to us now what those orphan fields are for now, us 6 months from now will be confused by this reuse and entanglement.

Reply:
It's package-private so it seemed a little easier to read. Just wanted to note this here for posterity 👍🏻